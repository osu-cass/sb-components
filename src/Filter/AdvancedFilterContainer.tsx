import * as React from "react";
import "../Styles/advanced-filter.less";
import { AdvancedFilter } from "./AdvancedFilter";
import {
  AdvancedFilterCategoryModel,
  AdvancedFiltersModel,
  FilterOptionModel,
  OptionTypeModel
} from "./AdvancedFilterModel";

export interface AdvancedFilterContainerProps {
  filterOptions: AdvancedFilterCategoryModel[];
  onClick: ( selected: AdvancedFilterCategoryModel[] ) => void;
}

export interface AdvancedFilterContainerState {
  expanded: boolean;
}

export class AdvancedFilterContainer extends React.Component<
  AdvancedFilterContainerProps,
  AdvancedFilterContainerState
  > {
  constructor ( props: AdvancedFilterContainerProps ) {
    super( props );

    this.state = {
      expanded: false
    };
  }

  handleClick = () => {
    this.setState( { expanded: !this.state.expanded } );
  };

  onSelect ( category: AdvancedFilterCategoryModel, option?: FilterOptionModel ) {
    const index = this.props.filterOptions.indexOf( category );
    const newFilters = [ ...this.props.filterOptions ];
    let newOptions: FilterOptionModel[] = [];

    //TODO Refactor
    if ( !option ) {
      // all pressed
      newOptions = newFilters[ index ].filterOptions.map( opt => ( {
        ...opt,
        isSelected: false
      } ) );
    } else {
      const optionIdx = newFilters[ index ].filterOptions.indexOf( option );
      if ( category.isMultiSelect ) {
        newOptions = newFilters[ index ].filterOptions.map( opt => ( { ...opt } ) );
      } else {
        newOptions = newFilters[ index ].filterOptions.map( opt => ( {
          ...opt,
          isSelected: false
        } ) );
      }

      newOptions[ optionIdx ].isSelected = !option.isSelected;
    }

    newFilters[ index ] = {
      ...newFilters[ index ],
      filterOptions: newOptions
    };

    this.props.onClick( newFilters );
  }

  resetFilters () {
    const newFilters = this.props.filterOptions;
    newFilters.forEach( cate => {
      cate.filterOptions.map( opt => ( opt.isSelected = false ) );
    } );
    this.props.onClick( newFilters );
  }

  hasActiveFilterIndicators () {
    let active = false;
    this.props.filterOptions.forEach( fil => {
      if ( !fil.disabled ) {
        fil.filterOptions.forEach( opt => {
          if ( opt.isSelected ) {
            active = true;
          }
        } );
      }
    } );
    return active;
  }

  renderFilterIndicators () {
    const tags: JSX.Element[] = [];

    this.props.filterOptions.forEach( fil => {
      if ( !fil.disabled ) {
        fil.filterOptions.forEach( opt => {
          if ( opt.isSelected ) {
            tags.push(
              <div className="filter-indicator" key={opt.key}>
                {opt.label}&nbsp;<span
                  onClick={() => this.onSelect( fil, opt )}
                  className="fa fa-times-circle fa-small"
                />
              </div>
            );
          }
        } );
      }
    } );

    return tags;
  }

  renderFilterHeader () {
    return (
      <div className="filter-header">
        <div className="filter-status">{this.renderFilterIndicators()}</div>
      </div>
    );
  }

  renderFilterBody () {
    const filterTags = this.props.filterOptions.map( ( fil, i ) => {
      return (
        <AdvancedFilter
          key={i}
          {...fil}
          selectedHandler={opt => this.onSelect( fil, opt )}
        />
      );
    } );

    return (
      <div
        className="filter-body"
        aria-live="polite"
        aria-relevant="additions removals"
      >
        {filterTags}
      </div>
    );
  }

  renderCollapsedFilterContainer = () => {
    const { filterOptions } = this.props;
    const className = this.state.expanded
      ? "fa fa-chevron-down"
      : "fa fa-chevron-right";
    const buttonText = this.state.expanded ? "Collapse " : "Expand ";
    return (
      <div className="filter-sub-header-container">
        <div className="filter-advanced-filter-header">
          <div className="filter-advanced-filter-title">
            <h2 style={{ color: "#63666A" }}>
              <span className="fa fa-tasks fa-lg" />&nbsp;Advanced Filters
            </h2>
            <span style={{ marginTop: "6px" }}>
              &nbsp;Click on an item to remove it from the list
            </span>
          </div>
          <div style={{ display: "flex", marginRight: "10px" }}>
            {this.hasActiveFilterIndicators() ? (
              <button
                onClick={() => this.resetFilters()}
                className="filter-button"
              >
                Reset Filters
              </button>
            ) : null}
            <button
              onClick={() => this.handleClick()}
              className="filter-button"
            >
              {buttonText}
              <span className={className} />
            </button>
          </div>
        </div>
        {filterOptions && filterOptions.length > 0
          ? this.renderFilterHeader()
          : null}
      </div>
    );
  };

  render () {
    let content = null;
    if ( this.state.expanded ) {
      content = (
        <div className="advanced-filter-container-expanded">
          {this.renderFilterBody()}
        </div>
      );
    }

    return (
      <div className="advanced-filter-container">
        {this.renderCollapsedFilterContainer()}
        {content}
      </div>
    );
  }
}
