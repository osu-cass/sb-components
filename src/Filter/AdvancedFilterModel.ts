export enum OptionType {
    inputBox,
    button
}

export interface AdvancedFilterOption {
    label: string;
    key: string;
    isSelected: boolean;
    type: OptionType;
}

export interface AdvancedFilterCategory {
    disabled: boolean;
    isMultiSelect: boolean;
    label: string;
    helpText: string;
    filterOptions: AdvancedFilterOption[];
    displayAllButton: boolean;
}

export interface TechType extends Subject { }

export interface Subject {
    code: string;
    label: string;
}

export interface AdvancedFilters {
    subjects: AdvancedFilterCategory;
    grades: AdvancedFilterCategory;
    techTypes: AdvancedFilterCategory;
}
