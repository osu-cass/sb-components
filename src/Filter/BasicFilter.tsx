import * as React from "react";
import {
  BasicFilterCategoryModel,
  FilterOptionModel,
  OptionTypeModel
} from "./FilterModels";
import { SelectOptionProps } from "../Select/SelectOption";
import { Select } from "../Select/Select";

export interface BasicFilterProps extends BasicFilterCategoryModel {
  selectedHandler: (data?: FilterOptionModel) => void;
}
/**
 * Renders a radio button or drop down list from
 * the FilterOptionModel passed in as props
 * @class BasicFilter
 * @extends {React.Component<BasicFilterProps, {}>}
 */
export class BasicFilter extends React.Component<BasicFilterProps, {}> {
  constructor(props: BasicFilterProps) {
    super(props);
  }

  findFilterOption(key: string) {
    return this.props.filterOptions.find(fil => fil.key === key);
  }

  /**
   * Renders JSX element of radio input category
   * @returns {JSX.Element} radio input category with selections
   */
  renderRadio(): JSX.Element {
    const { selectedHandler, label, filterOptions } = this.props;

    const radioOptions = filterOptions.map(fo => {
      return (
        <div className="radio" key={fo.key}>
          <label>
            <input
              checked={fo.isSelected}
              aria-checked={fo.isSelected}
              type="radio"
              name={label}
              value={fo.key}
              onChange={() => this.props.selectedHandler(fo)}
            />
            {fo.label}
          </label>
        </div>
      );
    });

    return (
      <label>
        {label}
        {radioOptions}
      </label>
    );
  }

  /**
   * Renders Select list for the category with default option
   * @returns {JSX.Element} Select React component
   */
  renderDropDown(): JSX.Element {
    const {
      selectedHandler,
      type,
      filterOptions,
      disabled,
      label,
      code
    } = this.props;
    const defaultValue = "default";

    const selected = filterOptions.find(fil => fil.isSelected === true);
    const selectedValue = selected ? selected.key : defaultValue;

    let selectOptions: SelectOptionProps[] = [];
    selectOptions.push({
      disabled,
      selected: selectedValue === defaultValue,
      label: "Select " + label,
      value: defaultValue
    });

    const filterSelectOptions = filterOptions.map(fo => {
      return {
        disabled,
        selected: selectedValue === fo.key,
        label: fo.label,
        value: fo.key
      };
    });

    selectOptions = selectOptions.concat(filterSelectOptions);

    return (
      <Select
        disabled={disabled}
        label={label}
        selected={selectedValue}
        options={selectOptions}
        onChange={val => selectedHandler(this.findFilterOption(val))}
        key={code}
        className={"input-sm med-text"}
      />
    );
  }

  /**
   * Renders the render Category based on the category type
   * @returns {(JSX.Element | undefined)} JSX element of category
   */
  renderCategory(): JSX.Element | undefined {
    const { type } = this.props;
    let content: JSX.Element | undefined;

    switch (type) {
      case OptionTypeModel.DropDown:
        content = this.renderDropDown();
        break;
      case OptionTypeModel.radioBtn:
        content = this.renderRadio();
        break;
      default:
        console.error("Invalid option", type);
        break;
    }

    return content;
  }

  /**
   * Renders an individual category
   * @returns default render method JSX Element
   */
  render() {
    let { label } = this.props;
    label = label.replace(/\ /g, "-");

    return (
      <div id={`${label}-bf`.toLocaleLowerCase()} className="bf-selection">
        {this.renderCategory()}
      </div>
    );
  }
}
