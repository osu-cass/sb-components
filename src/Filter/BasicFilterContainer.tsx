import * as React from "react";
import {
  BasicFilterCategoryModel,
  FilterOptionModel,
  FilterType
} from "./FilterModels";
import { BasicFilter } from "./BasicFilter";

/**
 * Properties interface for the BasicFilterContainer component.
 * @interface BasicFilterContainerProps
 * @member {BasicFilterCategoryModel[]} filterCategories
 * @member {(selected: BasicFilterCategoryModel[]) => void} onUpdateFilter
 * @member {boolean} containsAdvancedFilter
 * @member {() => void} handleAdvancedFilterExpand
 * @member {string?} filterId
 * @member {(() => void)?} resetHandler
 */
export interface BasicFilterContainerProps {
  filterCategories: BasicFilterCategoryModel[];
  onUpdateFilter: (
    selected: BasicFilterCategoryModel[],
    changed: FilterType
  ) => void;
  containsAdvancedFilter: boolean;
  handleAdvancedFilterExpand: () => void;
  filterId?: string;
  resetHandler?: () => void;
}
/**
 * State interface for the BasicFilterContainer component.
 * @interface BasicFilterContainerState
 * @member {boolean} expanded
 */
export interface BasicFilterContainerState {
  expanded?: boolean;
}
/**
 * The BasicFilterContainer is a menu of drop down lists and radio buttons
 * that filter Items based on BasicFilterCategoryModels and their corresponding
 * filter options. Can expand the Advanced filter as well if they are coupled together.
 * @class BasicFilterContainer
 * @extends {React.Component<BasicFilterContainerProps, BasicFilterContainerState>}
 */
export class BasicFilterContainer extends React.Component<
  BasicFilterContainerProps,
  BasicFilterContainerState
> {
  constructor(props: BasicFilterContainerProps) {
    super(props);
    this.state = {
      expanded: this.props.containsAdvancedFilter ? false : true
    };
    this.handleClick = this.handleClick.bind(this);
  }
  /**
   * Sets the isSelected property of the filter that the user selected
   * and sets the state of the parent component with the new filter categories.
   * @method onFilterSelect
   * @param {BasicFilterCategoryModel} category
   * @param {FilterOptionModel} [option]
   */
  onFilterSelect(
    category: BasicFilterCategoryModel,
    option?: FilterOptionModel
  ) {
    const index = this.props.filterCategories.indexOf(category);
    let { filterCategories } = this.props;
    if (!category.disabled) {
      if (option !== undefined) {
        let newOptions = filterCategories[index].filterOptions.slice();
        const optionIdx = filterCategories[index].filterOptions.indexOf(option);
        newOptions = filterCategories[index].filterOptions.map(opt => ({
          ...opt,
          isSelected: false
        }));

        newOptions[optionIdx].isSelected = !option.isSelected;
        filterCategories[index].filterOptions = newOptions;
        this.props.onUpdateFilter(filterCategories, category.code);
      }
    }
  }
  /**
   * Returns an array Basic Filter component for each of the filter categories
   * @method renderFilters
   */
  renderFilters() {
    const { filterCategories } = this.props;
    return filterCategories.map((fil, i) => {
      return (
        <BasicFilter
          key={i}
          {...fil}
          selectedHandler={opt => this.onFilterSelect(fil, opt)}
        />
      );
    });
  }

  /**
   * If the BasicFilter and AdvancedFilter Containers are used in conjunction with each other,
   * this method sets expands and collapses the AdvancedFilterContainer
   * @method handleClick
   */
  handleClick() {
    this.setState({ expanded: !this.state.expanded });
    this.props.handleAdvancedFilterExpand();
  }

  render() {
    const { filterCategories, containsAdvancedFilter, filterId } = this.props;
    const { expanded } = this.state;
    const id = filterId ? filterId : "";
    let advancedFilterButton = null;

    // if the component is being used in conjunction
    // with the AdvancedFilterContainer we handle expanding it here
    if (containsAdvancedFilter) {
      advancedFilterButton = (
        <div className="bf-button-container">
          <span>
            <button
              className="btn btn-default filter-button"
              onClick={this.props.resetHandler}
            >
              Reset
            </button>
          </span>
          <span>
            <button
              className="btn btn-default filter-button af-expand"
              onClick={this.handleClick}
            >
              Advanced Filters&nbsp;
              <span
                aria-hidden="true"
                className={`fa fa-chevron-${expanded ? "down" : "right"}`}
              />
            </button>
          </span>
        </div>
      );
    }
    let className = "bf-container section-light";
    if (expanded) {
      className = `${className} bf-expanded`;
    }

    return (
      <div id={id} className={className}>
        <div className="basic-filter">{this.renderFilters()}</div>
        {advancedFilterButton}
      </div>
    );
  }
}
