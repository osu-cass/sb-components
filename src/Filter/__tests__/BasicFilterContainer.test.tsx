import * as React from "react";
import { BasicFilterContainer } from "../BasicFilterContainer";
import { BasicFilter } from "../BasicFilter";
import { BasicFilterCategoryModel } from "../FilterModels";
import { mockBasicFilterCategories } from "../../../mocks/Filter/mocks";
import { shallow } from "enzyme";
import { Select } from "../../Select/Select";
import { AdvancedFilterContainer } from "../AdvancedFilterContainer";

describe("BasicFilterContainer", () => {
  const advFilExpand = jest.fn();

  const updateFilter = (filterCategories: BasicFilterCategoryModel[]) => {
    wrapper.setProps({ filterCategories });
  };

  const wrapper = shallow(
    <BasicFilterContainer
      filterCategories={mockBasicFilterCategories}
      onUpdateFilter={updateFilter}
      containsAdvancedFilter={false}
      handleAdvancedFilterExpand={() => {
        return;
      }}
    />,
    { lifecycleExperimental: true }
  );

  const wrapper1 = shallow(
    <BasicFilterContainer
      filterCategories={mockBasicFilterCategories}
      onUpdateFilter={updateFilter}
      containsAdvancedFilter={true}
      handleAdvancedFilterExpand={advFilExpand}
    />,
    { lifecycleExperimental: true }
  );

  it("has proper number of BasicFilters", () => {
    expect(wrapper.findWhere(node => node.type() === BasicFilter)).toHaveLength(
      2
    );
  });

  it("can select dropdown filter options", () => {
    const options = ["default", 7, 56, 960];
    options.forEach(opt => {
      wrapper
        .findWhere(node => node.type() === BasicFilter)
        .at(0)
        .dive()
        .findWhere(node => node.type() === Select)
        .simulate("change", { target: { value: opt } });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("hide select message if disabled", () => {
    const gradeDropDownModel = mockBasicFilterCategories[0];
    const view = shallow(
      // tslint:disable-next-line:no-empty
      <BasicFilter {...gradeDropDownModel} selectedHandler={() => {}} />
    );
    expect(view.findWhere(a => a.text().includes("Select")).length).toEqual(0);
    expect(view).toMatchSnapshot();
    const options = ["default", 7, 56, 960];
  });

  it("can select radio button filter options", () => {
    const options = ["ELA", "Math"];
    let i = 0;
    options.forEach(opt => {
      wrapper
        .findWhere(node => node.type() === BasicFilter)
        .at(1)
        .dive()
        .find("input")
        .at(i)
        .simulate("change", { target: { value: opt } });
      expect(wrapper).toMatchSnapshot();
      i++;
    });
  });

  it("resets filters on keyboard events", () => {
    const keyCodes = [0, 13, 32];
    keyCodes.forEach(key => {
      wrapper
        .findWhere(node => node.type() === BasicFilter)
        .at(0)
        .dive()
        .findWhere(node => node.type() === Select)
        .simulate("keyDown", { keyCode: key });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("expands the advanced filter", () => {
    expect(wrapper1).toMatchSnapshot();
    wrapper1.setState({ expanded: false });
    const expandButton = wrapper1
      .findWhere(node => node.type() === "button")
      .at(1);

    expandButton.simulate("click");
    expect(wrapper1).toMatchSnapshot();
    const advFilter = wrapper1.findWhere(
      node => node.type() === AdvancedFilterContainer
    );

    expect(advFilter).toBeDefined();
    expect(advFilExpand).toBeCalled();

    expandButton.simulate("click");
    expect(wrapper1).toMatchSnapshot();
  });
});
