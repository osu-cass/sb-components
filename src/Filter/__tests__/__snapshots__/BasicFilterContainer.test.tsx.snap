// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicFilterContainer can select dropdown filter options 1`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select dropdown filter options 2`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select dropdown filter options 3`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select dropdown filter options 4`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select radio button filter options 1`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select radio button filter options 2`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer expands the advanced filter 1`] = `
<div
  className="bf-container section-light"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
  <div
    className="bf-button-container"
  >
    <span>
      <button
        className="btn btn-default filter-button"
      >
        Reset
      </button>
    </span>
    <span>
      <button
        className="btn btn-default filter-button af-expand"
        onClick={[Function]}
      >
        Advanced Filters 
        <span
          aria-hidden="true"
          className="fa fa-chevron-right"
        />
      </button>
    </span>
  </div>
</div>
`;

exports[`BasicFilterContainer expands the advanced filter 2`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
  <div
    className="bf-button-container"
  >
    <span>
      <button
        className="btn btn-default filter-button"
      >
        Reset
      </button>
    </span>
    <span>
      <button
        className="btn btn-default filter-button af-expand"
        onClick={[Function]}
      >
        Advanced Filters 
        <span
          aria-hidden="true"
          className="fa fa-chevron-down"
        />
      </button>
    </span>
  </div>
</div>
`;

exports[`BasicFilterContainer expands the advanced filter 3`] = `
<div
  className="bf-container section-light"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
  <div
    className="bf-button-container"
  >
    <span>
      <button
        className="btn btn-default filter-button"
      >
        Reset
      </button>
    </span>
    <span>
      <button
        className="btn btn-default filter-button af-expand"
        onClick={[Function]}
      >
        Advanced Filters 
        <span
          aria-hidden="true"
          className="fa fa-chevron-right"
        />
      </button>
    </span>
  </div>
</div>
`;

exports[`BasicFilterContainer resets filters on keyboard events 1`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer resets filters on keyboard events 2`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer resets filters on keyboard events 3`] = `
<div
  className="bf-container section-light bf-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;
